skrub/_column_associations.py

```diff
@@ -247,3 +263,30 @@ def _compute_cramer(table, n_samples):
     stat = np.sqrt(chi_stat / (n_samples * np.maximum(min_dim, 1)))
     stat[min_dim == 0] = 0.0
     return stat
+
+
+def _compute_pearson(df):
+    """Compute the Pearson correlation coefficient for all pairs of columns.
+
+    This returns the symmetric matrix with shape (n cols, n cols) where entry
+    i, j contains the statistic for column i x column j.
+
+    """
+    # Replace categorical columns with np.nan to get a correlation matrix of shape
```

Comment:
I think we could simplify this a bit

- select numeric columns only
- use sbd.pearson_corr on those
- use pandas stack().reset_index() to convert to tall format
- or in polars case insert a column like `with_column(left_col=<the column names>)` then use `df.unpivot(index=left_col)` (only we would use a random name instead of `left_col` to have no name clash)
- then do the left join as you are doing in the caller


we would no longer use stack_symmetric_associations for this one I guess, but we would no longer need to do the filling with np.nan or the loops over columns. WDYT?

Reply:
Yes, I hesitated with this approach, but I saw avoiding manually dispatching between polars and pandas as an advantage, and reusing `stack_symmetric_associations` also makes things easier to understand IMO. Note that you wouldn't need to select numeric because `sbd.pearson_corr` does it for you. I agree that adding NaNs columns is a bit convoluted, though. WDYT?